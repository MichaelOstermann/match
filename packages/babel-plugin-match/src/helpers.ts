import * as t from "@babel/types"
import type { NodePath } from "@babel/core"
import type { LazyExpression, WellFormedObject } from "./types.js"

export class AbortError extends Error {}

export function isAbortError(value: unknown): value is AbortError {
    return value instanceof AbortError
}

export function* findCallExpressions(path: NodePath): Generator<NodePath<t.CallExpression>> {
    let currentPath = path.parentPath
    while (currentPath) {
        if (currentPath.isCallExpression()) {
            yield currentPath
        }
        currentPath = currentPath.parentPath
    }
}

// Plain object with static keys.
export function isWellFormedObject(value: t.Node): value is WellFormedObject {
    if (!t.isObjectExpression(value)) return false
    for (const property of value.properties) {
        if (!t.isObjectProperty(property)) return false
        if (property.computed) return false
        if (!t.isExpression(property.key)) return false
        if (!t.isExpression(property.value)) return false
    }
    return true
}

export function equals(a: t.Expression, b: t.Expression): boolean {
    if (a.type === "Identifier" && b.type === "Identifier") return a.name === b.name
    if (a.type === "StringLiteral" && b.type === "StringLiteral") return a.value === b.value
    if (a.type === "NumericLiteral" && b.type === "NumericLiteral") return a.value === b.value
    return false
}

export function getExpression(lazyExpression: LazyExpression): t.Expression {
    return typeof lazyExpression === "function"
        ? lazyExpression()
        : lazyExpression
}

export function createThrowExpression(value: t.Expression): t.Expression {
    // `Pattern matching error: no pattern matches value ${JSON.stringify(value)}`
    const message = t.templateLiteral([
        t.templateElement({ raw: "Pattern matching error: no pattern matches value " }),
        t.templateElement({ raw: "" }),
    ], [
        t.callExpression(t.memberExpression(t.identifier("JSON"), t.identifier("stringify")), [value]),
    ])

    // (() => { throw new Error(...) })()
    const error = t.newExpression(t.identifier("Error"), [message])
    const throwStatement = t.throwStatement(error)
    return t.callExpression(t.arrowFunctionExpression([], t.blockStatement([
        throwStatement,
    ])), [])
}
