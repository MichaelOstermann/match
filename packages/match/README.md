<div align="center">

<h1>match</h1>

**Zero-runtime exhaustive pattern matching.**

[Documentation](https://MichaelOstermann.github.io/match)

</div>
