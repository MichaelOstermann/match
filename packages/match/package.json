{
    "name": "@monstermann/match",
    "type": "module",
    "version": "0.1.0",
    "description": "Zero-runtime exhaustive pattern matching.",
    "author": "Michael Ostermann <michaelostermann@me.com>",
    "license": "MIT",
    "homepage": "https://MichaelOstermann.github.io/match",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/MichaelOstermann/match.git"
    },
    "bugs": {
        "url": "https://github.com/MichaelOstermann/match/issues"
    },
    "keywords": [
        "pattern",
        "matching",
        "pattern-matching",
        "exhaustive",
        "typescript",
        "match-with",
        "match",
        "switch"
    ],
    "sideEffects": false,
    "exports": {
        ".": {
            "types": "./dist/index.d.ts",
            "import": "./dist/index.js"
        }
    },
    "main": "./dist/index.js",
    "module": "./dist/index.js",
    "types": "./dist/index.d.ts",
    "files": [
        "dist"
    ],
    "scripts": {
        "bundle": "tsdown",
        "release": "bun publish --access public"
    },
    "dependencies": {
        "type-fest": "^4.41.0"
    }
}
