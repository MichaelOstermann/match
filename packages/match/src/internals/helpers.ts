import type { IsLiteral as IsNonNullableLiteral } from "type-fest"

export type IsTrue<T> = true extends T ? true : false
export type IfTrue<T, OnTrue, OnFalse> = true extends T ? OnTrue : OnFalse
export type Union<A, B> = [B] extends [A] ? A : [A] extends [B] ? B : A | B

// `type-fests` definition of literals + null and undefined.
export type IsLiteral<T> = IsNonNullableLiteral<T> extends true
    ? true
    : T extends null | undefined
        ? true
        : false

// @ts-expect-error ignore
export interface MatchError<i> { __: never }
