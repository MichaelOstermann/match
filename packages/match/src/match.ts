import { NonExhaustiveMatch } from "./NonExhaustiveMatch.js"
import type { Match } from "./internals/Match.js"

export function match<const T>(value: T): Match<T> {
    return new NonExhaustiveMatch(value) as unknown as Match<T>
}
