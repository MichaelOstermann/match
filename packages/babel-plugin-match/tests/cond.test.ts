import { describe, test } from "vitest"
import { expectSnapshot } from "./helpers.js"

describe("cond", () => {
    test("optimize #1", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(transform, true)
                .or(false)
        `)
    })

    test("optimize #2", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(() => value, true)
                .or(false)
        `)
    })

    test("optimize #3", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => v, true)
                .or(false)
        `)
    })

    test("optimize #4", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => v ? 1 : 2, true)
                .or(false)
        `)
    })

    test("optimize #5", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => v.foo, true)
                .or(false)
        `)
    })

    test("optimize #6", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => transform(v), true)
                .or(false)
        `)
    })

    test("optimize #7", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => transform(v.foo), true)
                .or(false)
        `)
    })

    test("optimize #8", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(function() { return value }, true)
                .or(false)
        `)
    })

    test("optimize #9", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(() => { return value }, true)
                .or(false)
        `)
    })

    test("optimize #10", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(function(v) { return v }, true)
                .or(false)
        `)
    })

    test("optimize #11", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => { return v }, true)
                .or(false)
        `)
    })

    test("optimize #12", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(function(v) { return v ? 1 : 2 }, true)
                .or(false)
        `)
    })

    test("optimize #13", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => { return v ? 1 : 2 }, true)
                .or(false)
        `)
    })

    test("optimize #14", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(function(v) { return v.foo }, true)
                .or(false)
        `)
    })

    test("optimize #15", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => { return v.foo }, true)
                .or(false)
        `)
    })

    test("optimize #16", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(function(v) { return transform(v) }, true)
                .or(false)
        `)
    })

    test("optimize #17", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => { return transform(v) }, true)
                .or(false)
        `)
    })

    test("optimize #18", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond(function(v) { return transform(v.foo) }, true)
                .or(false)
        `)
    })

    test("optimize #19", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(value)
                .cond((v) => { return transform(v.foo) }, true)
                .or(false)
        `)
    })

    test("optimize #20", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(transform, true)
                .or(false)
        `)
    })

    test("optimize #21", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(() => value, true)
                .or(false)
        `)
    })

    test("optimize #22", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => v, true)
                .or(false)
        `)
    })

    test("optimize #23", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => v ? 1 : 2, true)
                .or(false)
        `)
    })

    test("optimize #24", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => v.foo, true)
                .or(false)
        `)
    })

    test("optimize #25", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => transform(v), true)
                .or(false)
        `)
    })

    test("optimize #26", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => transform(v.foo), true)
                .or(false)
        `)
    })

    test("optimize #27", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(function() { return value }, true)
                .or(false)
        `)
    })

    test("optimize #28", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(() => { return value }, true)
                .or(false)
        `)
    })

    test("optimize #29", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(function(v) { return v }, true)
                .or(false)
        `)
    })

    test("optimize #30", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => { return v }, true)
                .or(false)
        `)
    })

    test("optimize #31", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(function(v) { return v ? 1 : 2 }, true)
                .or(false)
        `)
    })

    test("optimize #32", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => { return v ? 1 : 2 }, true)
                .or(false)
        `)
    })

    test("optimize #33", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(function(v) { return v.foo }, true)
                .or(false)
        `)
    })

    test("optimize #34", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => { return v.foo }, true)
                .or(false)
        `)
    })

    test("optimize #35", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(function(v) { return transform(v) }, true)
                .or(false)
        `)
    })

    test("optimize #36", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => { return transform(v) }, true)
                .or(false)
        `)
    })

    test("optimize #37", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond(function(v) { return transform(v.foo) }, true)
                .or(false)
        `)
    })

    test("optimize #38", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match({ foo: a, bar: b })
                .cond((v) => { return transform(v.foo) }, true)
                .or(false)
        `)
    })

    test("optimize #39", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(transform, true)
                .or(false)
        `)
    })

    test("optimize #40", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(() => value, true)
                .or(false)
        `)
    })

    test("optimize #41", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => v, true)
                .or(false)
        `)
    })

    test("optimize #42", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => v ? 1 : 2, true)
                .or(false)
        `)
    })

    test("optimize #43", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => v.foo, true)
                .or(false)
        `)
    })

    test("optimize #44", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => transform(v), true)
                .or(false)
        `)
    })

    test("optimize #45", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => transform(v.foo), true)
                .or(false)
        `)
    })

    test("optimize #46", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(function() { return value }, true)
                .or(false)
        `)
    })

    test("optimize #47", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(() => { return value }, true)
                .or(false)
        `)
    })

    test("optimize #48", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(function(v) { return v }, true)
                .or(false)
        `)
    })

    test("optimize #49", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => { return v }, true)
                .or(false)
        `)
    })

    test("optimize #50", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(function(v) { return v ? 1 : 2 }, true)
                .or(false)
        `)
    })

    test("optimize #51", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => { return v ? 1 : 2 }, true)
                .or(false)
        `)
    })

    test("optimize #52", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(function(v) { return v.foo }, true)
                .or(false)
        `)
    })

    test("optimize #53", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => { return v.foo }, true)
                .or(false)
        `)
    })

    test("optimize #54", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(function(v) { return transform(v) }, true)
                .or(false)
        `)
    })

    test("optimize #55", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => { return transform(v) }, true)
                .or(false)
        `)
    })

    test("optimize #56", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond(function(v) { return transform(v.foo) }, true)
                .or(false)
        `)
    })

    test("optimize #57", () => {
        expectSnapshot(`
            import { match } from "@monstermann/match"
            match(getValue())
                .cond((v) => { return transform(v.foo) }, true)
                .or(false)
        `)
    })
})
