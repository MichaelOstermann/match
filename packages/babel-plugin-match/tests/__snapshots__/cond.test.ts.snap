// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cond > optimize #1 1`] = `
"import { match } from "@monstermann/match";
transform(value) ? true : false;"
`;

exports[`cond > optimize #2 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #3 1`] = `
"import { match } from "@monstermann/match";
v ? true : false;"
`;

exports[`cond > optimize #4 1`] = `
"import { match } from "@monstermann/match";
(value ? 1 : 2) ? true : false;"
`;

exports[`cond > optimize #5 1`] = `
"import { match } from "@monstermann/match";
value.foo ? true : false;"
`;

exports[`cond > optimize #6 1`] = `
"import { match } from "@monstermann/match";
transform(value) ? true : false;"
`;

exports[`cond > optimize #7 1`] = `
"import { match } from "@monstermann/match";
transform(value.foo) ? true : false;"
`;

exports[`cond > optimize #8 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #9 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #10 1`] = `
"import { match } from "@monstermann/match";
v ? true : false;"
`;

exports[`cond > optimize #11 1`] = `
"import { match } from "@monstermann/match";
v ? true : false;"
`;

exports[`cond > optimize #12 1`] = `
"import { match } from "@monstermann/match";
(value ? 1 : 2) ? true : false;"
`;

exports[`cond > optimize #13 1`] = `
"import { match } from "@monstermann/match";
(value ? 1 : 2) ? true : false;"
`;

exports[`cond > optimize #14 1`] = `
"import { match } from "@monstermann/match";
value.foo ? true : false;"
`;

exports[`cond > optimize #15 1`] = `
"import { match } from "@monstermann/match";
value.foo ? true : false;"
`;

exports[`cond > optimize #16 1`] = `
"import { match } from "@monstermann/match";
transform(value) ? true : false;"
`;

exports[`cond > optimize #17 1`] = `
"import { match } from "@monstermann/match";
transform(value) ? true : false;"
`;

exports[`cond > optimize #18 1`] = `
"import { match } from "@monstermann/match";
transform(value.foo) ? true : false;"
`;

exports[`cond > optimize #19 1`] = `
"import { match } from "@monstermann/match";
transform(value.foo) ? true : false;"
`;

exports[`cond > optimize #20 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)({
  foo: a,
  bar: b
});"
`;

exports[`cond > optimize #21 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #22 1`] = `
"import { match } from "@monstermann/match";
v ? true : false;"
`;

exports[`cond > optimize #23 1`] = `
"import { match } from "@monstermann/match";
(_value => (_value ? 1 : 2) ? true : false)({
  foo: a,
  bar: b
});"
`;

exports[`cond > optimize #24 1`] = `
"import { match } from "@monstermann/match";
a ? true : false;"
`;

exports[`cond > optimize #25 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)({
  foo: a,
  bar: b
});"
`;

exports[`cond > optimize #26 1`] = `
"import { match } from "@monstermann/match";
transform(a) ? true : false;"
`;

exports[`cond > optimize #27 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #28 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #29 1`] = `
"import { match } from "@monstermann/match";
v ? true : false;"
`;

exports[`cond > optimize #30 1`] = `
"import { match } from "@monstermann/match";
v ? true : false;"
`;

exports[`cond > optimize #31 1`] = `
"import { match } from "@monstermann/match";
(_value => (_value ? 1 : 2) ? true : false)({
  foo: a,
  bar: b
});"
`;

exports[`cond > optimize #32 1`] = `
"import { match } from "@monstermann/match";
(_value => (_value ? 1 : 2) ? true : false)({
  foo: a,
  bar: b
});"
`;

exports[`cond > optimize #33 1`] = `
"import { match } from "@monstermann/match";
a ? true : false;"
`;

exports[`cond > optimize #34 1`] = `
"import { match } from "@monstermann/match";
a ? true : false;"
`;

exports[`cond > optimize #35 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)({
  foo: a,
  bar: b
});"
`;

exports[`cond > optimize #36 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)({
  foo: a,
  bar: b
});"
`;

exports[`cond > optimize #37 1`] = `
"import { match } from "@monstermann/match";
transform(a) ? true : false;"
`;

exports[`cond > optimize #38 1`] = `
"import { match } from "@monstermann/match";
transform(a) ? true : false;"
`;

exports[`cond > optimize #39 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)(getValue());"
`;

exports[`cond > optimize #40 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #41 1`] = `
"import { match } from "@monstermann/match";
(_value => v ? true : false)(getValue());"
`;

exports[`cond > optimize #42 1`] = `
"import { match } from "@monstermann/match";
(_value => (_value ? 1 : 2) ? true : false)(getValue());"
`;

exports[`cond > optimize #43 1`] = `
"import { match } from "@monstermann/match";
(_value => _value.foo ? true : false)(getValue());"
`;

exports[`cond > optimize #44 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)(getValue());"
`;

exports[`cond > optimize #45 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value.foo) ? true : false)(getValue());"
`;

exports[`cond > optimize #46 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #47 1`] = `
"import { match } from "@monstermann/match";
value ? true : false;"
`;

exports[`cond > optimize #48 1`] = `
"import { match } from "@monstermann/match";
(_value => v ? true : false)(getValue());"
`;

exports[`cond > optimize #49 1`] = `
"import { match } from "@monstermann/match";
(_value => v ? true : false)(getValue());"
`;

exports[`cond > optimize #50 1`] = `
"import { match } from "@monstermann/match";
(_value => (_value ? 1 : 2) ? true : false)(getValue());"
`;

exports[`cond > optimize #51 1`] = `
"import { match } from "@monstermann/match";
(_value => (_value ? 1 : 2) ? true : false)(getValue());"
`;

exports[`cond > optimize #52 1`] = `
"import { match } from "@monstermann/match";
(_value => _value.foo ? true : false)(getValue());"
`;

exports[`cond > optimize #53 1`] = `
"import { match } from "@monstermann/match";
(_value => _value.foo ? true : false)(getValue());"
`;

exports[`cond > optimize #54 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)(getValue());"
`;

exports[`cond > optimize #55 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value) ? true : false)(getValue());"
`;

exports[`cond > optimize #56 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value.foo) ? true : false)(getValue());"
`;

exports[`cond > optimize #57 1`] = `
"import { match } from "@monstermann/match";
(_value => transform(_value.foo) ? true : false)(getValue());"
`;
